### PR TITLE
[gulp-javascript-obfuscator] Add type definitions

### PR DESCRIPTION
**Pull request by MysteryBlokHed**
As requested in #58443, adds type definitions for [gulp-javascript-obfuscator](https://www.npmjs.com/package/gulp-javascript-obfuscator).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test gulp-javascript-obfuscator`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## jsforce ([<kbd>unpkg</kbd>](https://unpkg.com/browse/jsforce@latest/))
was missing the following properties:
1. addListener
1. on
1. once
1. removeListener
1. off

<details>
<summary>as well as these 12 other properties...</summary>
<p>removeAllListeners, setMaxListeners, getMaxListeners, listeners, rawListeners, emit, listenerCount, prependListener, prependOnceListener, eventNames, VERSION, require</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against b609f49e81051b750a8e28287b0422bb05576ebc
</p>

File: types/gulp-javascript-obfuscator/package.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "private": true,
+    "dependencies": {
+        "javascript-obfuscator": ">=1"
```
Comment (peterblazejewicz) [inline types/gulp-javascript-obfuscator/package.json:4]: can this be fixed to some recent version? thx!

Comment (MysteryBlokHed) [reply types/gulp-javascript-obfuscator/package.json]: I changed it to `^3.0.0`, the latest major release

**Review by peterblazejewicz — APPROVED** [review]
LGTM!

**Comment by MysteryBlokHed** [comment]
Ready to merge

**Pull request merged**